tests/test_dir_traversal.py

```diff
@@ -257,6 +267,484 @@ class DirectoryTraversalVector(Generic[T]):
             ],
         ],
     ),
+    DirectoryTraversalVector(
+        "ignored_dot_dirs__are_not_traversed",
+        given=[
+            File(".venv/sub/file"),
+            File("dir/.venv/sub/file"),
+            File("dir/foo.py"),
+        ],
+        ignore_patterns=[".*"],  # ignore all paths that start with "."
+        expect=[
+            ExpectedTraverseStep(".", subdirs=["dir"]),  # ignored_subdirs=[".venv"]
+            ExpectedTraverseStep("dir", files=["foo.py"]),  # ignored_subdirs=[".venv"]
+        ],
+    ),
+    #
+    # Testing ignore patterns
+    #
+    # The following tests are based on the pattern format rules listed in
+    # `git help ignore` (https://git-scm.com/docs/gitignore#_pattern_format).
+    #
+    # A blank line matches no files, so it can serve as a separator for
+    # readability.
+    DirectoryTraversalVector(
+        "gitignore_parsing__disregard_blank_lines",
+        given=[File("dir/file")],
+        ignore_patterns=["", ""],
```

Comment:
I am not sure I follow why `", "` is a blank line

Reply:
There are two blank lines: `""`, and `""`...